src/nbl/CMakeLists.txt

```diff
@@ -575,6 +575,123 @@ if(_NBL_COMPILE_WITH_GLI_)
 	target_include_directories(Nabla PUBLIC ${THIRD_PARTY_SOURCE_DIR}/gli)
 endif()
 
+# DXC
+if(_NBL_COMPILE_WITH_DXC_)
+	add_dependencies(Nabla dxc)
+	target_include_directories(Nabla PRIVATE 
+		"${DXC_BINARY_DIR_GE}/tools/clang/tools/dxcompiler"
+		"${DXC_SOURCE_DIR}/tools/clang/tools/dxcompiler"
+		"${DXC_SOURCE_DIR}/tools/clang/include"
+		"${DXC_BINARY_DIR_GE}/tools/clang/include"
+		"${DXC_SOURCE_DIR}/include/llvm/llvm_assert"
+		"${DXC_BINARY_DIR_GE}/include"
+		"${DXC_SOURCE_DIR}/include"
+		"${DXC_BINARY_DIR_GE}/include/dxc/Tracing"
+		"${DXC_BINARY_DIR_GE}/utils/version"
+	)
+	target_link_libraries(Nabla INTERFACE "${DXC_DXCOMPILER_LIB}")
+	
+	############################################################################
+	# list files you want to compile with DXC in _NBL_ENABLE_DXC_COMPILE_TESTS_
+	# once we have guarantee all of shaders in Nabla can be compiled with DXC,
+	# we can simply grep all of those files/use existing vars holding paths to
+	# shaders and remove manual adding the files bellow
+	###################################################
+	
+	if(_NBL_ENABLE_DXC_COMPILE_TESTS_)
+		set(_NBL_DXC_CT_INCLUDE_ROOT_ "${NBL_ROOT_PATH}/include") # Nabla include root path for shaders
+		set(_NBL_DXC_CT_BINARY_BIN_ "${CMAKE_CURRENT_BINARY_DIR}/DXC/CT") # compile-test directory
+		
+		if(_NBL_DXC_COMPILE_TESTS_ENABLE_CMAKE_LOG_)
+			message("${_NBL_DXC_CT_PREFIX_CMAKE_LOG_DEBUG_} _NBL_DXC_CT_INCLUDE_ROOT_ = '${_NBL_DXC_CT_INCLUDE_ROOT_}'")
+			message("${_NBL_DXC_CT_PREFIX_CMAKE_LOG_DEBUG_} _NBL_DXC_CT_BINARY_BIN_ = '${_NBL_DXC_CT_BINARY_BIN_}'")
+		endif()
+		
+		# this is how you add file for compilation test
+		set(_NBL_DXC_CT_SHADER_FILES_ 
+			"${_NBL_DXC_CT_INCLUDE_ROOT_}/nbl/builtin/hlsl/barycentric/utils.hlsl"
```

Comment:
@nahim-yay this is how you want to test HLSL shaders and files that should be compiled with DXC

Reply:
We need to be able to spec `lib_6_7` target or ps/vs/gs/etc target, entry point and spirv out

Reply:
resolved